### ISSUE TITLE
Disallow invalid URLs

### ISSUE THREAD

**Issue by AdityaAnand1** [linkedtales/scrapedin#81]
Right now it throws an exception if you give scrapedin an invalid url.

Instead, there should be URL sanity check before the headless page is ever opened. Only valid profile URLs should be allowed.

**Comment by alex-kj-chin** [comment]
Also, it would be helpful for the error message to contain the url.

**Comment by leonardiwagner** [comment]
TLDR; we don't check about "valids URL"

----
Long explanation: 
> Right now it throws an exception if you give scrapedin an invalid url.

@AdityaAnand1 @alex-kj-chin I think you guys are telling about "LinkedIn website changed and scrapedin can't read basic data." error right?

First, it isn't about it being an invalid URL, actually, scrapedin try to check **any** URL and throws that error just because couldn't find data on the URL.

Why we don't have a check about being a "valid linkedin profile" ? Because the way we do now is checking for an specific CSS selector that only exists on a Linkedin profile.

What happened when Linkedin changes that specific CSS while the rest works? Scrapedin thrown and error even though we could read the data. 

-----
Suggestion: 
- In your application, you should handle which URL is gonna to be scraped.. if this error happens.. then you know which URL is invalid.
- Send a suggestion to track invalid URLs without the fragility to break the rest of the program.

**Comment by AdityaAnand1** [comment]
Thanks for the response @leonardiwagner, I think this issue is a symptom of an underlying cause. I think in general the same error message:

>LinkedIn website changed and scrapedin can't read basic data

In all situations is not very helpful, including the reason for failure would be excellent.

- Maybe the website isn't linkedin.com
- Maybe the CSS changed
- Maybe the user isn't logged in

And so many others..

Right now these are all handled with:

>LinkedIn website changed and scrapedin can't read basic data

Having more specific error throwing inside scrapedin, would be extremely helpful.

- I might want to see what happened if the user got auto-logged out
- I might want to fix the scrapedin CSS selectors if something was changed
- I might want to fix my data if the website was incorrect

**Comment by leonardiwagner** [comment]
> I might want to fix the scrapedin CSS selectors if something was changed
I might want to fix my data if the website was incorrect

Alright, what's possible to do is:
- If the profile page CSS selector is not found, we save that state and:
  - Situation A: If we still collect data.. it was a false negative... carries on normally.
  - Situation B: If we don't have any information, throw an error saying it's probably about not being on a profile page.

>  I might want to see what happened if the user got auto-logged out

That's really a good idea to try to find if we're logged while trying to scrape, it may solve some problems related to inconsistency about the session.. we can use a "retry N times" the login process if we lose the session by some reason.

>  scrapedin is a bit of a black box for the rest of the application,

Ok, I got it, I'm going to improve these feedbacks to help it. It's just hard to have the assurance to thrown feedbacks relying on states that can be changed suddenly, but we're going to improve indeed based in your feedback. Thank you. More suggestions, PRs, etc.. are welcomed, thanks :smile:
